### ISSUE TITLE
len_zero false negative for default implementations of `is_empty` in traits

### ISSUE THREAD

**Issue by leops** [rust-lang/rust-clippy#8384]
### Summary

The `len_zero` lint is only emitted when the `is_empty` method is implemented explicitly for a type, and is not raised when said method is inherited as a default implementation from a trait

### Lint Name

len_zero

### Reproducer

I tried this code:

```rust
trait ListTrait {
    fn slice(&self) -> &[u8];
    
    fn len(&self) -> usize {
        self.slice().len()
    }
    
    fn is_empty(&self) -> bool {
        self.slice().is_empty()
    }
}

pub struct Wrapper(Vec<u8>);

impl ListTrait for Wrapper {
    fn slice(&self) -> &[u8] {
        &self.0
    }
}

pub fn check_empty(wrapper: &Wrapper) -> bool {
    wrapper.len() == 0
}
```

I expected to see this happen:
The `wrapper.len() == 0` should raise the `len_zero` lint

Instead, this happened:
No warning were emitted

### Version

```text
rustc 1.58.0 (02072b482 2022-01-11)
binary: rustc
commit-hash: 02072b482a8b5357f7fb5e5637444ae30e423c40
commit-date: 2022-01-11
host: x86_64-pc-windows-msvc
release: 1.58.0
LLVM version: 13.0.0
```
